drivers/platform/stm32/stm32_spi.c

```diff
@@ -283,15 +281,35 @@ int32_t stm32_spi_transfer(struct no_os_spi_desc *desc,
 	}
 
 	for (uint32_t i = 0; i < len; i++) {
-		rx_cnt = 0;
-		tx_cnt = 0;
 
 		/* Assert CS */
 		gdesc->port->BSRR = NO_OS_BIT(sdesc->chip_select->number) << 16;
 
 		if(msgs[i].cs_delay_first)
 			no_os_udelay(msgs[i].cs_delay_first);
 
+#ifndef SPI_SR_TXE
+		/* Some STM32 families have different naming for
+		   SPI_SR_TXE, SPI_SR_RXNE and SPIx->DR. In that case, simply
+		   use the HAL API for SPI transmission, which is generic
+		   for all STM32 families. */
+		if (msgs[i].tx_buff && msgs[i].rx_buff)
+			ret = HAL_SPI_TransmitReceive(&sdesc->hspi, msgs[i].tx_buff, msgs[i].rx_buff,
+						      msgs[i].bytes_number, HAL_MAX_DELAY);
+		else {
+			uint8_t * data = msgs[i].tx_buff ? msgs[i].tx_buff : msgs[i].rx_buff;
+			ret = HAL_SPI_TransmitReceive(&sdesc->hspi, data, data,
```

Comment:
Won't this try to overwrite the `tx_buff` array? The behavior prior to this commit was to not do so.

Reply:
thank you for pointing this out, fixed this behavior with the latest changes